### PR TITLE
feat: nft card

### PR DESCRIPTION
**Pull request by d0c0q**

### CODE REVIEW

File: contracts/nft/OtterCardNFT.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import '@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol';
+
+contract OtterCardNFT is ERC721, ERC721URIStorage, Ownable {
+    string public cardURI;
+    uint256 public tokenIdCount;
+
+    mapping(address => bool) public whitelist;
+    mapping(address => uint256) public claimed;
```
Comment (otterclamking) [inline contracts/nft/OtterCardNFT.sol:13]: can we use `mapping(address => bool)` ?

**Review by otterclamking — COMMENTED** [review]
Where is the deploy script?

File: contracts/nft/OtterCardNFT.sol (hunk 1 above)
Comment (d0c0q) [reply contracts/nft/OtterCardNFT.sol:13]: uint256 for token id

File: contracts/nft/OtterCardNFT.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        string memory cardURI_
+    ) ERC721(name_, symbol_) {
+        cardURI = cardURI_;
+    }
+
+    function _burn(uint256 tokenId)
```
Comment (otterclamking) [inline contracts/nft/OtterCardNFT.sol:23]: Do we need to override this function or can just remove this part?

Comment (d0c0q) [reply contracts/nft/OtterCardNFT.sol:23]: it's needed.

ERC721 and ERC721URIStorage both implement this function, so you need to  explicitly make it call by inheritance order.

**Review by otterclamking — APPROVED** [review]

**Pull request merged**